pandas/tests/indexes/test_multi.py

```diff
@@ -3291,3 +3291,37 @@ def test_duplicate_multiindex_labels(self):
         with pytest.raises(ValueError):
             ind.set_levels([['A', 'B', 'A', 'A', 'B'], [2, 1, 3, -2, 5]],
                            inplace=True)
+
+    testdata = [
+        (pd.MultiIndex.from_product([[0, 1], [1, 0]]), pd.Series(range(4)), 4),
+        (pd.MultiIndex.from_product([[0, 1]]), pd.Series(range(2)), 2),
+    ]
+
+    @pytest.mark.parametrize("midx,idx,count", testdata)
+    def test_multiindex_compare(self, midx, idx, count):
+        # GH 21149 - change in 'def cmp_method()'
+        expected = pd.Series([True]).repeat(count)
+        expected.reset_index(drop=True, inplace=True)
+        result = pd.Series(midx == midx)
```

Comment:
not really clear what you are testing, this is way too dense. put blank lines and a comment in between cases.